sm/plugins/tedge_apt_plugin/src/module_check.rs

```diff
@@ -0,0 +1,72 @@
+use crate::error::InternalError;
+use std::path::PathBuf;
+use std::process::{Command, Stdio};
+
+/// check that module_name is in file path
+pub fn filepath_has_extension(file_path: &str) -> bool {
+    let pb = PathBuf::from(file_path);
+    match pb.extension() {
+        Some(extension) => extension == "deb",
+        None => false,
+    }
+}
+
+pub struct PackageMetadata {
+    metadata: Option<String>,
+}
+
+impl PackageMetadata {
+    pub fn try_new(file_path: &str) -> Result<Self, InternalError> {
+        let metadata = String::from_utf8(Self::get_module_metadata(file_path)?)?;
+        Ok(Self {
+            metadata: Some(metadata),
+        })
+    }
+
+    pub fn metadata_contains_all(&self, patterns: &[&str]) -> bool {
+        for pattern in patterns {
+            if !self.metadata_contains(pattern) {
+                return false;
+            }
+        }
+        true
+    }
+
+    fn metadata_contains(&self, pattern: &str) -> bool {
+        if let Some(lines) = &self.metadata {
+            return lines.contains(pattern);
+        }
+        false
+    }
+
+    fn get_module_metadata(file_path: &str) -> Result<Vec<u8>, InternalError> {
+        Ok(Command::new("dpkg")
+            .arg("-I")
+            .arg(file_path)
+            .stdout(Stdio::piped())
+            .output()?
+            .stdout)
+    }
+}
+
+pub fn validate_package(file_path: &str, contain_args: &[&str]) -> Result<String, InternalError> {
+    let package_metadata = PackageMetadata::try_new(file_path).unwrap();
+
+    if package_metadata.metadata_contains_all(contain_args) {
+        if !filepath_has_extension(file_path) {
+            let file_path = create_link_with_extension(file_path)?;
+            return Ok(file_path);
```

Comment:
There is something strange here. One checks the file first then we create a symlink to have a file with the extension expected by apt.

Is this ".deb" extension actually required?

Reply:
Unfortunately it is when using `apt-get`:
```shell
$ sudo apt-get install ./.tmp/rolldice_1.16-1build1_amd64

Reading package lists... Done
E: Unsupported file ./.tmp/rolldice_1.16-1build1_amd64 given on commandline
```

whereas for file with extension:
```shell
$ sudo apt-get install ./.tmp/rolldice_1.16-1build1_amd64.deb 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  rolldice
0 upgraded, 1 newly installed, 0 to remove and 56 not upgraded.
Need to get 0 B/9628 B of archives.
After this operation, 31.7 kB of additional disk space will be used.
Selecting previously unselected package rolldice.
(Reading database ... 73170 files and directories currently installed.)
Preparing to unpack .../rolldice_1.16-1build1_amd64.deb ...
Unpacking rolldice (1.16-1build1) ...
Setting up rolldice (1.16-1build1) ...
Processing triggers for man-db (2.9.1-1) ...
```